module/Photo/src/Photo/Mapper/Photo.php

```diff
@@ -30,12 +30,79 @@ public function __construct(EntityManager $em)
     }
 
     /**
+     * Returns the next photo in the album to display
+     * 
+     * @param \Photo\Model\Photo $photo 
+     */
+    public function getNextPhoto($photo)
+    {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->select('a')
+                ->from('Photo\Model\Photo', 'a')
+                ->where('a.id > ?1 AND a.album = ?2');
+        $qb->setParameter(1, $photo->getId());
+        $qb->setParameter(2, $photo->getAlbum());
+        $qb->addOrderBy('a.id', 'ASC');
+        $qb->setMaxResults(1);
+        $res = $qb->getQuery()->getResult();
+        return empty($res) ? null : $res[0];
+    }
+
+    /**
+     * Returns the previous photo in the album to display
+     * 
+     * @param \Photo\Model\Photo $photo 
+     */
+    public function getPreviousPhoto($photo)
+    {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->select('a')
+                ->from('Photo\Model\Photo', 'a')
+                ->where('a.id < ?1 AND a.album = ?2');
+        $qb->setParameter(1, $photo->getId());
+        $qb->setParameter(2, $photo->getAlbum());
+        $qb->addOrderBy('a.id', 'DESC');
+        $qb->setMaxResults(1);
+        $res = $qb->getQuery()->getResult();
+        return empty($res) ? null : $res[0];
+    }
+
+    /**
+     * Returns a random photo from the specified album
+     * 
+     * @param Photo\Model\Album $album
+     * @return Photo\Model\Photo
+     */
+    public function getRandomPhoto($album)
+    {
+        $qb = $this->em->createQueryBuilder();
+
+        /**
+         * This is not exactly the right way to do this. There is no way to just
+         * get a random row, unless we use a raw sql query.
+         * The doctrine team refuses to implement any 
+         * random functionality. (http://www.doctrine-project.org/jira/browse/DDC-950)
+         * 
+         */
+        $qb->select('a')
+                ->from('Photo\Model\Photo', 'a')
+                ->where('a.album = ?1');
+        $qb->setParameter(1, $album);
+        $qb->setMaxResults(20);
+        $res = $qb->getQuery()->getResult();
+        return empty($res) ? null : $res[array_rand($res)];
```

Comment:
@nickygerritsen Dit kan nog steeds langzaam zijn afhankelijk van hoe groot rand()%cnt uitvalt. Voor offset gaat mysql volgens mij eerst langs alle rijen van 0 t/m offset voordat het iets returned. Een statische cover foto die van te voren gegenereerd is zal altijd sneller zijn.

@kokx Als we het allemaal eens zijn met dat idee, pas het dan gerust aan in de spec. Dan open ik ook wel even een extra issue daarvoor. Ik zie overigens net dat zoiets als mijn idee al een beetje in de spec beschreven staat eigenlijk.